### PR TITLE
Add version checker to CI

### PR DESCRIPTION
**Pull request by LuRsT**

### CODE REVIEW

**Comment by LuRsT** [comment]
@unmadeworks/backend Let me know how you feel about that bash script

**Comment by LuRsT** [comment]
@BenVosper, it's apparently "the" way to do it :man_shrugging:. I'd love to find a better way tho

File: .circleci/check_version.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
```
Comment (mocoso) [inline .circleci/check_version.sh]: Can you add a bit more of an explanation of what this commit achieves to the commit message - i.e. what is the behaviour change?

**Comment by mocoso** [comment]
> @BenVosper, it's apparently "the" way to do it 🤷‍♂. I'd love to find a better way tho

@LuRsT: If you are slightly uncomfortable with this approach, but it is the 'standard way' to do this in Python how about you add this information to the commit and perhaps point at some references to show how you discovered it was the 'standard way'?

File: .circleci/check_version.sh (hunk 1 above)
Comment (LuRsT) [reply .circleci/check_version.sh]: Done

File: .circleci/check_version.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# Collect the name and version from `pyproject.toml`
+NAME=$(cat pyproject.toml | grep name | sed 's/"//g' | awk '{print $3}')
+VERSION=$(cat pyproject.toml | grep version | sed 's/"//g' | awk '{print $3}')
+
+# Running `pip install packaganame==` returns this in STDERR
+#   ERROR: Could not find a version that satisfies the requirement packagename== (from versions: 0.1.0, 0.1.1, 0.1.2)
+
+# |& pipes into STDOUT both STDOUT and STDERR
+pip install "$NAME==" |& grep $VERSION
+
```
Comment (pssuils) [inline .circleci/check_version.sh]: Let me see if I follow correctly the code:
- Extract name and version of the package on the repo
- Try to install `package==version`
- If you can install it means that the version is incorrect?

Not sure if it's my less experience on `bash` or that some of the algorith will need to be explicit either code or commit message.

Also, being this on `pr` what would happen with multiple PRs opened at the same time? 

Why the name is extracted from the project file, are you expected to be change the name of the package?

Comment (LuRsT) [reply .circleci/check_version.sh]: > Try to install package==version

When you run the command `pip install requests==` (notice the lack of a version number), you get a list of all published versions, that `grep` checks if the version we have, is there.

If there is a version in the output, it means it has been published already.

> Not sure if it's my less experience on bash or that some of the algorith will need to be explicit either code or commit message.

I tried to be explicit on the comments, if the explanation above is clearer, I'm happy to change the comment appropriately.

> Also, being this on pr what would happen with multiple PRs opened at the same time?

The same thing that happens now with the other packages, you need to bump each PR.

> Why the name is extracted from the project file, are you expected to be change the name of the package?

The script shouldn't be tied to the package, nor the package name.

Comment (pssuils) [reply .circleci/check_version.sh]: Can you explain further : 

> The script shouldn't be tied to the package, nor the package name.

Advantages of it for example?

**Comment by LuRsT** [comment]
Merging this before leaving, @pssuils let me know if I was clear in my message, I'm happy to change everything, or anyone should be able to as well :wink:

**Pull request merged**